sidecar_acceptor.go

```diff
@@ -589,25 +566,25 @@ func (a *SidecarAcceptor) handleServerMessage(
 // a bid order) the tasks are simplified compared to normal bid order execution.
 //
 // NOTE: The lock must be held when calling this method.
-func (a *SidecarAcceptor) matchPrepare(pendingBatch *order.Batch,
-	msg *auctioneerrpc.OrderMatchPrepare) (*order.Batch, error) {
+func (a *SidecarAcceptor) matchPrepare(
+	msg *auctioneerrpc.OrderMatchPrepare) error {
 
 	// Parse and formally validate what we got from the server.
 	batch, err := order.ParseRPCBatch(msg)
 	if err != nil {
-		return nil, fmt.Errorf("unable to parse batch: %v", err)
+		return fmt.Errorf("unable to parse batch: %v", err)
 	}
 
 	sdcrLog.Infof("Received PrepareMsg for batch=%x, num_orders=%v",
 		batch.ID[:], len(batch.MatchedOrders))
 
 	// If there is still a pending batch around from a previous iteration,
 	// we need to clean up the pending channels first.
-	if pendingBatch != nil {
-		if err := a.removeShims(pendingBatch); err != nil {
-			return nil, fmt.Errorf("unable to cleanup previous "+
-				"batch: %v", err)
+	if a.pendingBatch != nil {
+		if err := a.removeShims(a.pendingBatch); err != nil {
```

Comment:
So this is part of the core fix here? That we wipe the batch after we remove the shim?

Reply:
Yes, the fix was merged in another PR (already in master). Usually we want to remove the shims of the previous batch, but it was possible to end up in a state were there was not previous batch stored and some shims were created